### PR TITLE
proxy_* params were removed from class apt

### PR DESCRIPTION
**Pull request by mhaskel**
Add them to PPA since they were being used there, and add a placeholder
example for setting up the proxy files.

### CODE REVIEW

File: manifests/ppa.pp [hunk 1]
```diff
@@ -5,6 +5,8 @@
   $options        = $::apt::ppa_options,
   $package_name   = $::apt::ppa_package,
   $package_manage = false,
+  $proxy_host     = undef,
+  $proxy_port     = 8080,
```
Comment (daenney) [inline manifests/ppa.pp]: Mmm, I've started disliking having multiple variables that are an obvious pair/entity as multiple switches on the class and instead have taken to doing something like this:

```puppet
class apt::params {
  $proxy = { 'host': undef, 'port': 8080 }
}

define apt::ppa (
  $proxy = {}
)  {
  $_proxy = merge($apt::proxy, $proxy)
}
```

This allows the user to still only specify `proxy => { 'host': '127.0.0.1' }` and get the port for free.

What do you think?

Comment (mhaskel) [inline manifests/ppa.pp]: @daenney I don't hate that, but I am a little confused with your hash syntax. I've only remembering seeing puppet hashes with `=>` as the separator between key/value (and also [this](https://docs.puppetlabs.com/puppet/latest/reference/lang_datatypes.html#hashes)). Is this something new?

Comment (daenney) [inline manifests/ppa.pp]: No, I've just been doing too much of non-Ruby/Puppet as of late. I meant hash rockets.

Comment (mhaskel) [inline manifests/ppa.pp]: Ok, cool, just wanted to make sure I didn't miss something new :) I'll get this updated.

**Comment by daenney** [comment]
Mmm, we actually have no tests covering this change.

**Pull request merged**